static/js/services/contact-save.js

```diff
@@ -28,12 +28,18 @@ angular.module('inboxServices').service('ContactSave',
 
     function prepareSubmittedDocsForSave(schema, original, submitted) {
       var doc = prepare(submitted.doc);
-      var repeated = prepareRepeatedDocs(submitted.doc, submitted.repeats);
 
       return prepareAndAttachSiblingDocs(schema, submitted.doc, original, submitted.siblings)
         .then(function(siblings) {
-          doc.parent = ExtractLineage(doc.parent);
-          doc.contact = ExtractLineage(doc.contact);
+          siblings.concat(doc).map(function(item) {
```

Comment:
The map function doesn't return anything so I think it should be just a `forEach`.

Also it's slightly wasteful and slightly harder to read to create a new array just to iterate over it then throw it away. Consider pulling out a function that extracts the lineage of parent and contact and then...

```js
someAwesomeName(doc);
sibilings.forEach(someAwesomeName);
```

Reply:
Also, `prepareAndAttachSiblingDocs` is doing extraction on the doc.parent and doc.contact which is probably not needed any more.

I'm now wondering why we don't do the extraction just before the `bulkDocs` call (around line 150). The doc, the siblings and the repeated docs are all returned, so we could just iterate over extracting parent and contact for all of them, then doing the save. Anyway, just a thought...

Reply:
> slightly harder to read 

Interesting! It's a trivial point and I think they are similar in either direction, but if I had to pick one I thought was more readable it would be coming the inputs to be put through one logic block, not extracting the logic block and calling it multiple times.